pandas/io/formats/format.py

```diff
@@ -633,20 +633,30 @@ def _calc_max_cols_fitted(self) -> Optional[int]:
 
     def _calc_max_rows_fitted(self) -> Optional[int]:
         """Number of rows with data fitting the screen."""
-        if not self._is_in_terminal():
-            return self.max_rows
-
-        _, height = get_terminal_size()
-        if self.max_rows == 0:
-            # rows available to fill with actual data
-            return height - self._get_number_of_auxillary_rows()
-
-        max_rows: Optional[int]
-        if self._is_screen_short(height):
-            max_rows = height
+        if self._is_in_terminal():
+            _, height = get_terminal_size()
+            if self.max_rows == 0:
+                # rows available to fill with actual data
+                return height - self._get_number_of_auxillary_rows()
+
+            max_rows: Optional[int]
+            if self._is_screen_short(height):
+                max_rows = height
+            else:
+                max_rows = self.max_rows
         else:
             max_rows = self.max_rows
 
+        return self._adjust_max_rows(max_rows)
+
+    def _adjust_max_rows(self, max_rows: Optional[int]) -> Optional[int]:
```

Comment:
I don't think it is worth creating a separate function for those few lines, I think it fits fine in the function above, as it was before

Reply:
I thought that some explanation should be added why there are still some changes to max_rows and comparison with frame length. Could be a comment, but a separate function with the docstring is better, isn't it.